### ISSUE TITLE
Avoid streaming remote file content to generate hashes for packages

### ISSUE THREAD

**Issue by viveklak** [quiltdata/quilt#1650]
While creating packages out of remote content with 100k+ files, generating hashes forces the streaming of all the files which is excruciatingly slow. Perhaps we could consider a "trusted" mode for package creation such that the S3 Etag is used as a weaker replacement for the SHA256 sum that quilt currently uses?

**Comment by akarve** [comment]
Code snippet to repro. This appears to be streaming the entire package over the wire?

```
p = quilt3.Package.browse("CORD-19/2020-06-12", registry="s3://ai2-semanticscholar-cord-19")

p.set('pubs.vega.json', "content/pubs.vega.json")
  p.set('CORD19.ipynb', 'content/CORD19.ipynb')
  summarize = [s for s in summarize if s in p]
  p.set('quilt_summarize.json', entry=summarize)
  print(f"Initial push for {package_name} to {bucket}/{path}")
  p.push(package_name,
         bucket,
         dest=os.path.join(bucket, path),
         selector_fn=local_only,
         message="Adding new files")

**Comment by viveklak** [comment]
Here is the updated code. The use case is that there are 100k+ files in the `path` already in S3 and we are converting them to a package and adding a handful of files. I can clearly see all the files being fetched back serially from S3 as part of the hashing process.
```
b = quilt3.Bucket(bucket)
p.set_dir('/', os.path.join(bucket, path))
p.set('pubs.vega.json', "content/pubs.vega.json")
p.set('CORD19.ipynb', 'content/CORD19.ipynb')
summarize = [s for s in summarize if s in p]
p.set('quilt_summarize.json', entry=summarize)
print(f"Initial push for {package_name} to {bucket}/{path}")
p.push(package_name,
                   bucket,
                   dest=os.path.join(bucket, path),
                   selector_fn=local_only(bucket),
                   message=f"Expanded archive {a}")

def local_only(bucket: str) -> Callable[[str, 'quilt3.PackageEntry'], bool]:
    def f(_: str, entry: 'quilt3.PackageEntry') -> bool:
        return entry.get().startswith('file://')
    return f
```

**Comment by akarve** [comment]
To be resolved in #2782
